ui/components/app/assets/asset-list/asset-list.tsx

```diff
@@ -67,26 +73,63 @@ export type AssetListProps = {
   showTokensLinks?: boolean;
 };
 
+const TokenListContainer = React.memo(
+  ({ onClickAsset }: Pick<AssetListProps, 'onClickAsset'>) => {
+    const trackEvent = useContext(MetaMetricsContext);
+
+    const balance = useSelector(getMultichainSelectedAccountCachedBalance);
+
+    const {
+      currency: primaryCurrency,
+      numberOfDecimals: primaryNumberOfDecimals,
+    } = useUserPreferencedCurrency(PRIMARY, {
+      ethNumberOfDecimals: 4,
+      shouldCheckShowNativeToken: true,
+    });
+
+    const [, primaryCurrencyProperties] = useCurrencyDisplay(balance, {
+      numberOfDecimals: primaryNumberOfDecimals,
+      currency: primaryCurrency,
+    });
```

Comment:
This logic differs from cherry-pick since we also merged a large refactor.

https://github.com/MetaMask/metamask-extension/pull/29886